Doc/whatsnew/3.13.rst

```diff
@@ -2005,231 +2006,510 @@ CPython Bytecode Changes
+  :c:func:`!_PyThreadState_UncheckedGet`.
+  (Contributed by Victor Stinner in :gh:`108867`.)
+
+* Add the :c:func:`PyType_GetFullyQualifiedName` function
+  to get the type's fully qualified name.
+  This is equivalent to:
+
+  .. code-block:: python
+
+     if isinstance(type.__module__, str) and type.__module__ != 'builtins':
+         return f'{type.__module__}.{type.__qualname__}'
+     else:
+         return type.__qualname__
+
+  (Contributed by Victor Stinner in :gh:`111696`.)
+
+* Add the :c:func:`PyType_GetModuleName` function
+  to get the type's module name.
+  This is equivalent to getting the ``type.__module__`` attribute.
+  (Contributed by Eric Snow and Victor Stinner in :gh:`111696`.)
+
+* Add the :c:func:`PyUnicode_EqualToUTF8AndSize`
+  and :c:func:`PyUnicode_EqualToUTF8` functions
+  to compare a Unicode object with a :c:expr:`const char*` UTF-8 encoded string
+  and ``1`` if they are equal or ``0`` otherwise.
   These functions do not raise exceptions.
   (Contributed by Serhiy Storchaka in :gh:`110289`.)
 
-* Add :c:func:`PyThreadState_GetUnchecked()` function: similar to
-  :c:func:`PyThreadState_Get()`, but don't kill the process with a fatal error
-  if it is NULL. The caller is responsible to check if the result is NULL.
-  Previously, the function was private and known as
-  ``_PyThreadState_UncheckedGet()``.
-  (Contributed by Victor Stinner in :gh:`108867`.)
+* Add the :c:func:`PyWeakref_GetRef` function
+  as an alternative to  :c:func:`PyWeakref_GetObject`
+  that returns a :term:`strong reference`
+  or ``NULL`` if the referent is no longer live.
+  (Contributed by Victor Stinner in :gh:`105927`.)
 
-* Add :c:func:`PySys_AuditTuple` function: similar to :c:func:`PySys_Audit`,
-  but pass event arguments as a Python :class:`tuple` object.
-  (Contributed by Victor Stinner in :gh:`85283`.)
+* Add fixed variants of functions which silently ignore errors:
+
+  * :c:func:`PyObject_HasAttrWithError` replaces :c:func:`PyObject_HasAttr`.
+  * :c:func:`PyObject_HasAttrStringWithError`
+    replaces :c:func:`PyObject_HasAttrString`.
+  * :c:func:`PyMapping_HasKeyWithError` replaces :c:func:`PyMapping_HasKey`.
+  * :c:func:`PyMapping_HasKeyStringWithError`
+    replaces :c:func:`PyMapping_HasKeyString`.
 
-* :c:func:`PyArg_ParseTupleAndKeywords` now supports non-ASCII keyword
-  parameter names.
+  The new functions return ``-1`` for errors
+  and the standard ``1`` for true and ``0`` for false.
+
+  (Contributed by Serhiy Storchaka in :gh:`108511`.)
+
+
+Changed C APIs
+--------------
+
+* The *keywords* parameter of :c:func:`PyArg_ParseTupleAndKeywords`
+  and :c:func:`PyArg_VaParseTupleAndKeywords`
+  now has type :c:expr:`char * const *` in C
+  and :c:expr:`const char * const *` in C++,
+  instead of :c:expr:`char **`.
+  This makes these functions compatible with arguments
+  of type :c:expr:`const char * const *`, :c:expr:`const char **`,
+  or :c:expr:`char * const *` in C++ and :c:expr:`char * const *` in C
```

Comment:
Is "``char * const *`` in C and ``char * const *`` in C++" a typo? It repeats the type twice for C and C++ with no changes, which seems odd.

Reply:
Hmm, `char * const *` seems to be the syntax of cpp.